### PR TITLE
Backup scripts and Documentation

### PR DESCRIPTION
**Pull request by jjasghar**
- Updated README
- Added Backup scripts
- Updated metadata.rb
- Added integration tests for cron and backup script

### CODE REVIEW

**Comment by drrk** [comment]
Looks, good, but the backup script ideally wants to send `save-off` followed by `save-all flush`  to the minecraft console prior to the backup, and then `save-on` after completion.  This ensures that the backup is complete and good, ie has the current state of the world, and an auto save doesn't happen during the copy corrupting the backup.

note that `save-all flush` will freeze the server while the chunks flush to disk, so its also worth sending a message to the online players too with `say <message>`

**Comment by drrk** [comment]
I also note, that currently you don't start the server with a named pipe to the console, so sending the commands isn't easy atm :(

File: test/integration/default/serverspec/default_spec.rb [hunk 1]
```diff
@@ -13,6 +13,15 @@
     it { should exist }
   end
 
+  describe file('/root/backup_minecraft.sh') do
+    its(:content) { should match /$DATE-backup-minecraft.tar.gz/ }
+    it { should exist }
+  end
+
+  describe cron do
+    it { should have_entry '00 02 * * * /root/backup_minecraft.sh' }
```
Comment (ncerny) [inline test/integration/default/serverspec/default_spec.rb:22]: You should stub the node attributes here, or else hard-coded the cron settings in the recipe.

File: templates/default/backup_minecraft.sh.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (ncerny) [inline templates/default/backup_minecraft.sh.erb:1]: Should we set a path for the tar to go to?  Or do you want it to always be the script's working path?

File: recipes/default.rb [hunk 3]
```diff
@@ -68,6 +68,19 @@
   mode '0644'
 end
 
+template '/root/backup_minecraft.sh' do
+  source 'backup_minecraft.sh.erb'
+  owner 'root'
+  group 'root'
+  mode '0755'
```
Comment (ncerny) [inline recipes/default.rb:76]: Why would anyone other than root need to execute this?

File: recipes/default.rb [hunk 4]
```diff
@@ -68,6 +68,19 @@
...
+end
+
+cron "backup_minecraft" do
+  hour "#{node['minecraft_basic']['backup_hour']}"
+  minute "#{node['minecraft_basic']['backup_min']}"
+  command "/root/backup_minecraft"
```
Comment (ncerny) [inline recipes/default.rb:82]: missing extension `.sh`

**Pull request closed without merging**